component/authenticator/src/test/java/org/wso2/carbon/extension/identity/authenticator/passwordpolicy/test/PasswordChangeEnforcerOnExpirationTest.java

```
@@ -0,0 +1,355 @@
+
+    @Mock
+    private AuthenticatorConfig authenticatorConfig;
+
+    @Mock
+    private ApplicationAuthenticator applicationAuthenticator;
+
+    @Mock
+    private LocalApplicationAuthenticator localApplicationAuthenticator;
+
+    @Mock
+    private UserStoreManager userStoreManager;
+
+    @Mock
+    private UserRealm userRealm;
+
+    @Mock
+    private RealmService realmService;
+
+    @BeforeMethod
+    public void setUp() throws Exception {
+        passwordChangeEnforcerOnExpiration = new PasswordChangeEnforcerOnExpiration();
+        initMocks(this);
+    }
+
+    @AfterMethod
+    public void tearDown() throws Exception {
+    }
+
+    @Test
+    public void testCanHandleTrue() {
+        Assert.assertEquals(passwordChangeEnforcerOnExpiration.canHandle(httpServletRequest), true);
+    }
+
+    @Test
+    public void testGetFriendlyName() {
+        Assert.assertEquals(passwordChangeEnforcerOnExpiration.getFriendlyName(),
+                PasswordChangeEnforceConstants.AUTHENTICATOR_NAME);
+    }
+
+    @Test
+    public void testGetName() {
+        Assert.assertEquals(passwordChangeEnforcerOnExpiration.getName(),
+                PasswordChangeEnforceConstants.AUTHENTICATOR_NAME);
+    }
+
+    @Test
+    public void testGetContextIdentifier() {
+        when(httpServletRequest.getParameter(PasswordChangeEnforceConstants.STATE)).thenReturn("abc");
+        Assert.assertEquals(passwordChangeEnforcerOnExpiration.getContextIdentifier(httpServletRequest),
+                "abc");
+    }
+
+
+    @Test
+    public void testGetUsername() throws Exception {
+        when(context.getSequenceConfig()).thenReturn(sequenceConfig);
+        when(sequenceConfig.getStepMap()).thenReturn(mockedMap);
+        when(mockedMap.get(anyObject())).thenReturn(stepConfig);
+        Whitebox.invokeMethod(passwordChangeEnforcerOnExpiration, "getUsername", context);
+    }
+
+    @Test
+    public void testUpdateAuthenticatedUserInStepConfig() throws Exception {
```

Should we throw a generic exception here?